src/txmempool.h

```diff
@@ -298,7 +298,7 @@ class CompareTxMemPoolEntryByAncestorFee
 
         if (f1 == f2)
         {
-            return a.GetTx().GetHash() < b.GetTx().GetHash();
+            return a.GetTime() < b.GetTime();
```

Comment:
Yes, gives a perfect fill in the block when for instance we have a long package which is greater than the maxblocksize (or the remaining amount of block to be filled) and all the fees for each transaction are equal.  This sorting does two things.  This way we end up finding the part of the package that first perfectly into the block, because we don't bail out early due to pacakge insertion failures, while also preserving some sense of fairness that the first transaction that arrived in the mempool has priority over the one that followed.  (Remember in BU and also ABC there is no longer any random delay in forwarding inv's anymore.)